docs/data/migration/usage-with-material-ui-v5-v6/usage-with-material-ui-v5-v6.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Usage with Material UI v5 and v6
+---
+
+# Usage with Material UI v5/v6
+
+<p class="description">This guide describes the changes needed to use the Data Grid with Material UI v5/v6.</p>
+
+## Package layout changes
+
+MUI X v8 packages have been updated to use the [Node.js `exports` field](https://nodejs.org/api/packages.html#exports), following [Material v7 package layout changes](https://mui.com/system/migration/upgrade-to-v7/#package-layout).
+
+MUI X v8 packages are compatible with Material UI v7 out of the box.
+We encourage upgrading to Material UI v7 to take advantage of better ESM support.
+
+Material UI v6 and v5 can still be used but require some additional steps if you are importing the packages in a Node.js environment.
+
+### General recommendations
+
+Make sure to pass `require` [condition](https://nodejs.org/api/cli.html#-c-condition---conditionscondition) when importing MUI X packages in Node.js environment:
+
+```bash
+node --conditions=require index.mjs
+```
+
+### Vite
+
+Update Vite configuration to pass the `require` condition:
+
+```ts
+// vite.config.js
+import { defineConfig } from 'vite';
+
+export default defineConfig({
+  ssr: {
+    resolve: {
+      externalConditions: ['require'],
```

Comment:
Maybe we can link to https://vite.dev/config/ssr-options#ssr-resolve-externalconditions
and also mention something about the callout in that doc?

> When using this option, make sure to run Node with [--conditions flag](https://nodejs.org/docs/latest/api/cli.html#-c-condition---conditionscondition) with the same values in both dev and build to get a consistent behavior.
>
> For example, when setting ['node', 'custom'], you should run NODE_OPTIONS='--conditions custom' vite in dev and NODE_OPTIONS="--conditions custom" node ./dist/server.js after build.

Reply:
Good call, I didn't notice that 👍🏻